src/services/memberProfile/repository.ts

```diff
@@ -0,0 +1,49 @@
+import { v4 } from 'uuid';
+
+import { Member } from '@graasp/sdk';
+
+import { AppDataSource } from '../../plugins/datasource';
+import { MemberProfile } from './entities/profile';
+import { IMemberProfile } from './types';
+
+interface ProfileFilter {
+  visibility: boolean;
+}
+const MemberProfileRepository = AppDataSource.getRepository(MemberProfile).extend({
+  async createOne(
+    args: IMemberProfile & {
+      member: Member;
+    },
+  ): Promise<MemberProfile> {
+    const { bio, visibility = false, facebookLink, linkedinLink, twitterLink, member } = args;
+
+    const id = v4();
+
+    const memberProfile = await this.create({
+      id,
+      bio,
+      visibility,
+      facebookLink,
+      linkedinLink,
+      twitterLink,
+      member,
+    });
+    await this.insert(memberProfile);
+    return memberProfile;
+  },
+
+  async getByMemberId(memberId: string, filter?: ProfileFilter): Promise<MemberProfile> {
```

Comment:
Can you add an additional check on memberId? Because typeorm will return you the first item if `memberId` is  undefined..